integrationtest/mapper/orm-realbackend/src/test/java/org/hibernate/search/integrationtest/mapper/orm/realbackend/mapping/DecimalScaleMappingIT.java

```diff
@@ -39,11 +45,32 @@ public void testFailingWithHint() {
 				);
 	}
 
+	@Test
+	public void testScaleDerivedFromJpaColumn() {
```

Comment:
Good point on the new test - will move it.
As for the original one, I've left the questions for today 😄. I think that it wouldn't work like that anymore. Meaning the error and the hint worked because there was no info on the ID property. But now that we are including ID - the scale is set, even if the `@Column` is not there explicitly. And now it fails with:
```
    Hibernate ORM mapping: 
        type 'org.hibernate.search.integrationtest.mapper.orm.realbackend.mapping.DecimalScaleMappingIT$FailingEntity': 
            path '.id': 
                index 'failing': 
                    failures: 
                      - HSEARCH400570: Invalid index field type: decimal scale '2' is positive. The decimal scale of BigInteger fields must be zero or negative.
```
we probably can also use some hint here, but on the other hand, it looks more like a problem with schema generation... so it seems that this test would only make sense with the standalone mapper?

Reply:
> And now it fails with:

But wait, that failure *is* a problem, and apparently not the user's fault. So I assume the default scale is wrong for `BigInteger` in Hibernate ORM? Is this a problem with metadata or does ORM map things wrong on the database side too? Is it solved in ORM 6? Does this affect only IDs or also normal properties (I just looked and it seems we don't test mapping of `BigInteger` without an explicit scale, so I can't be sure...)

> it seems that this test would only make sense with the standalone mapper?

Feel free to only test the hint with the Standalone POJO Mapper if you can't reproduce the problem with Hibernate ORM, yes.

But still the problem above needs to be investigated.

Reply:
> > And now it fails with:
> 
> But wait, that failure _is_ a problem, and apparently not the user's fault. So I assume the default scale is wrong for `BigInteger` in Hibernate ORM? Is this a problem with metadata or does ORM map things wrong on the database side too? Is it solved in ORM 6? Does this affect only IDs or also normal properties (I just looked and it seems we don't test mapping of `BigInteger` without an explicit scale, so I can't be sure...)

Feel free to create a separate ticket to investigate this, but please have a look :)

Reply:
Yup, that's what worried me yesterday, I think on the ORM side, it creates something like `numeric(19,2)` or similar. Will take a closer look 😃

Reply:
Thanks. You might want to test with ORM 6, because of lots of things changed in that area, so this might be a bug that is specific to Hibernate ORM 5.

Reply:
It looks to be fine with 6.2. With 5 if the mapping is:
```java
@Id
@GenericField
@Column(name = "id", nullable = false, precision = 18)
private BigInteger id;
```
then everything seems fine:
```
sqltype = numeric(18,0)
```
But as soon as there's no `@column` or it is defined without precision:
```java 
@Id
@GenericField
private BigInteger id;
//or:
@Id
@GenericField
@Column(name = "id", nullable = false)
private BigInteger id;
```
then it leads to:
```
sqltype = numeric(19,2)
//or
sqltype = numeric(19,2)
```

should we open a ticket for ORM 5? and then for our tests ... I guess we could add an explicit column for the time (in addition, we can remove it in the orm6 patches)

Reply:
> should we open a ticket for ORM 5?

It's probably not worth it. I doubt it will get fixed, it looks like a long-standing bug, and it's already solved in 6.

> and then for our tests

Just use `BigDecimal`, maybe? Seems easier.